game-headed/src/main/java/org/triplea/game/client/ui/javafx/TripleA.java

```diff
@@ -51,6 +55,10 @@ public void start(final Stage stage) throws Exception {
     scene.getStylesheets().add(FxmlManager.STYLESHEET_MAIN.toString());
     mainMenu = addRootContent(new MainMenuPane(this));
     setupStage(stage, scene);
+    // Don't invoke Swing if headless (for example in tests)
```

Comment:
If we're running tests locally, as a dev, will the game frame be rendered?

It's probably a best practice to allow tests to run in the background and not do things like steal screen focus. It does take 3-5 minutes to run the full TripleA verification, it's too time inefficient to sit and wait for that, I'm very often doing other work while waiting for the verification run to complete. FWIW, It would be very bothersome to not have the tests simply run in the background.

It would be great to have a clean solution here, I'm not sure what that would be. Simply having a test constant to disallow the rendering during test might be a better fix.

Reply:
> If we're running tests locally, as a dev, will the game frame be rendered?

No, the only reason I'm creating a JFrame is for the LookAndFeel (thinking about it, probably unecessary) and all of the icons to be correct. Also there's a lot of code depending on a present (non-null) frame to display loading screens etc.

So the frame is only created, but never displayed/made visible.

Reply:
A comment indicating that I think would be helpful. The current comment is all about tests, at first glance it looks like we're rendering the frame. Perhaps a TODO statement with the next steps would make it clear there is work in progress here.